gridstatus/pjm.py

```diff
@@ -257,29 +267,107 @@ def get_lmp(
                 'HUB', 'EHV', 'TIE', 'RESIDUAL_METERED_EDC'.
 
         """
+        if locations == "hubs":
+            locations = self.hub_node_ids
+
+        if location_type:
+            location_type = location_type.upper()
+            if location_type not in self.location_types:
+                raise ValueError(
+                    f"location_type must be one of {self.location_types}",
+                )
+
         if date == "latest":
-            """Currently only supports DAY_AHEAD_HOURlY"""
+            """Supports DAY_AHEAD_HOURLY, REAL_TIME_5_MIN"""
             return self._latest_lmp_from_today(
                 market=market,
                 locations=locations,
                 location_type=location_type,
                 verbose=verbose,
             )
 
-        if locations == "hubs":
-            locations = self.hub_node_ids
+        recent_threshold = pd.Timestamp.now(tz=self.default_timezone) - pd.Timedelta(
+            days=DV_LMP_RECENT_NUM_DAYS,
+        )
+        dv_df = None
+        if utils._handle_date(date, tz=self.default_timezone) >= recent_threshold:
+            dv_df = self._get_lmp_via_dv(
+                date,
+                market,
+                end=end,
+                locations=locations,
+                location_type=location_type,
+                verbose=verbose,
+            )
+
+        # Figure out whether we need to call the JSON API
+        if utils.is_today(date, tz=self.default_timezone):
+            fetch_json = market in (Markets.DAY_AHEAD_HOURLY,)
+        else:
+            fetch_json = True
+
+        json_df = None
+        if fetch_json:
+            json_df = self._get_lmp_via_pjm_json(
+                date,
+                market,
+                end=end,
+                locations=locations,
+                location_type=location_type,
+                verbose=verbose,
+            )
+
+        dfs = [dv_df, json_df]
+        dfs = [df for df in dfs if df is not None]
 
+        # deduplicate data, choosing the _src=json when there are duplicates
+        # in (time, market, location name)
+        df = self._df_deduplicate(
```

Comment:
The original implementation was definitely too large to inline. This version is more compact and I bet it could be reused elsewhere if we need to blend data sources. In an ideal world I think this method belongs in a `df_utils.py`.

I think having a slightly generalized version (as opposed to hardcoding in `_src` values being `dv` or `json`) helps maintainability - let me know if you feel strongly about inlining this though.